python/tank/platform/bundle.py

```diff
@@ -394,6 +382,27 @@ def import_module(self, module_name):
         
         return sys.modules[mod_name]
 
+    def get_project_cache_location(self, project_id):
+        """
+        Gets the bundle's cache-location path for the given project id.
+
+        :param project_id:  The project Entity id number.
+
+        :returns:           str
+        """
+        # this method is memoized for performance since it is being called a lot!
+        if self.__cache_location.get(project_id) is None:
+            pc_id = self.__tk.pipeline_configuration.get_shotgun_id()
+            
+            self.__cache_location[project_id] = self.__tk.execute_core_hook_method(
+                constants.CACHE_LOCATION_HOOK_NAME,
+                "bundle_cache",
+                project_id=project_id,
+                pipeline_configuration_id=pc_id,
+                bundle=self
+            )
+        
+        return self.__cache_location[project_id]
```

Comment:
on a similar note, is it enough to key off just project id - do we need to add a pipeline config id into the mix here?